functions/src/helpers.ts

```diff
@@ -2,3 +2,103 @@
 export const extractListFromFilename = fileName => {
   return fileName.match(/^[^-]*[^ -]/g)[0];
 };
+
+
+
+//////////////////////////////
+//
+//  Converts a column number into its equivalent A1 notation
+//      Example: 3 --> c
+//
+//////////////////////////////
+const columnToLetter = (column) => {
+  var temp, letter = '';
+  while (column > 0) {
+    temp = (column - 1) % 26;
+    letter = String.fromCharCode(temp + 65) + letter;
+    column = (column - temp - 1) / 26;
+  }
+  return letter;
+}
+
+export const buildSheetIndex = async (sheets, spreadsheetId) => {
+  const currentSheets = await sheets.spreadsheets.get({
+      spreadsheetId
+  });
+
+  const sheetTitles = [];
+  let columnsIndex = {}; // { ColumnName: columnNumber }
+
+  for (let i = 0; i < currentSheets.data.sheets.length; i++) {
+      let { title } = currentSheets.data.sheets[i].properties;
+      let { rowCount, columnCount } = currentSheets.data.sheets[i].properties.gridProperties;
+
+
+      const result = await sheets.spreadsheets.values.get({
+          spreadsheetId,
+          range: `${title}!1:1` // get all the cells in the first row `Column headers`
+      });
+
+      if (!result.data.values || result.data.values.length === 0) {
+          console.log(`No data found in sheet: ${title}`);
+          continue;
+      }
+  
+      // Getting only the sheets that have the required attributes
+      const row = result.data.values[0] 
+      if (row.indexOf('Beginning') > -1 && row.indexOf('Ending') > -1) {
```

Comment:
This is specific to your particular function, it's not proper to put this in helpers. Helpers should provide generic code that can be used for any other spreadsheet with another columns.